dist/bash-completion/tpm2-tools/tpm2_completion.bash

```diff
@@ -0,0 +1,153 @@
+#!/bin/sh
+# bash completion for tmp2-tools                                 -*- shell-script -*-
+
+
+_tpm2_tools()
+{
+
+
+
+    local cur prev words cword
+    _init_completion || return
+    local common_options=(-h --help -v --version -V --verbose -Q --quiet -Z --enable-errata -T --tcti=)
+    local aux1=$( ${COMP_WORDS[0]} -h summary 2>/dev/null )
+    local aux2=$( echo "${aux1}" | tr "[]|" " " | awk '{if(NR>2)print}' | tr " " "\n")
+    suggestions=("${aux2[@]}" "${common_options[@]}") #generate all the opts for the tool
+    local halg=(0x4 sha1 0xB sha256 0xC sha384 0xD sha512 0x12 sm3_256)
+    local public_object_alg=(0x1 rsa 0x8 keyedhash 0x23 ecc 0x25 symcipher)
+    local signing_alg=(0x5 hmac 0x14 rsassa 0x16 rsapss 0x18 ecdsa 0x1A ecdaa 0x1B sm2 0x1C ecschnorr)
+    local signing_schemes=(0x5 hmac 0x14 rsassa 0x15 rsaes 0x16 rsapss 0x17 oeap)
+    local tcti_opts=(device: mssim: abrmd:)
+
+
+    case $prev in
+      -g)
```

Comment:
I see that `tpm2_create -g` and tab completes the algorithms which is great. Do you know if it would be possible to also do the same for the long option? For example `tpm2_create --halg=` make to autocomplete?

If you don't know it's OK, that can always be added later, I just thought it would be nice to support both short and long options.